### PR TITLE
User Management - add new organization membership apis

### PR DESCRIPTION
**Pull request by jbvolta**
## Description
User Management - add new organization membership apis

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: lib/UserManagement.php [hunk 1]
```diff
@@ -180,7 +181,32 @@ public function addUserToOrganization($userId, $organizationId)
             true
         );
 
-        return Resource\User::constructFromResponse($response);
+        return Resource\OrganizationMembership::constructFromResponse($response);
+    }
+
+    /**
+     * Add a user to an organization.
```
Comment (amygdalama) [inline lib/UserManagement.php:188]: got some copy/pasta here

File: lib/UserManagement.php [hunk 2]
```diff
@@ -164,12 +164,13 @@ public function deleteUser($userId)
      *
      * @return \WorkOS\Resource\Response
      */
-    public function addUserToOrganization($userId, $organizationId)
+    public function createOrganizationMembership($userId, $organizationId)
     {
-        $userOrganizationPath = "users/{$userId}/organizations";
+        $userOrganizationPath = "user_management/organization_memberships";
```
Comment (amygdalama) [inline lib/UserManagement.php:169]: nit: i would rename this, maybe `organizationMembershipsPath`? (applies to the other methods here too)

File: lib/UserManagement.php [hunk 3]
```diff
@@ -205,7 +231,50 @@ public function removeUserFromOrganization($userId, $organizationId)
             true
         );
 
-        return Resource\User::constructFromResponse($response);
+        return $response;
+    }
+
+    /**
+     * List memberships for user and organization.
```
Comment (amygdalama) [inline lib/UserManagement.php:238]: Maybe `List organization memberships` ?

File: lib/UserManagement.php (hunk 2 above)
Comment (jbvolta) [reply lib/UserManagement.php:169]: Called it $path, simplier

File: lib/UserManagement.php [hunk 4]
```diff
@@ -164,48 +164,118 @@ public function deleteUser($userId)
...
             $params,
             true
         );
 
-        return Resource\User::constructFromResponse($response);
+        return Resource\OrganizationMembership::constructFromResponse($response);
+    }
+
+    /**
+     * Get Organization Memberships.
```
Comment (amygdalama) [inline lib/UserManagement.php:188]:
```suggestion
     * Get an Organization Membership.
```

**Review by amygdalama — APPROVED** [review]
one more minor comment wording tweak, otherwise looks good. thank you!

**Pull request merged**